### DISCUSSION TITLE
Training an LLM

### DISCUSSION THREAD

**Discussion by ballcoach12** [eclipse-langium/langium#1568] [Q&A] [answered]
Sorry if this is off-topic, but I was wondering if anyone has experimented with training an LLM to recognize a DSL.  The first release of our language is almost ready and we are roadmapping future tools that we will likely want to provide to customers.  I am not an AI expert at all, so I am trying to figure out of training a model for a DSL is technically out of reach for my team.  If anyone has done this, I'd be interested to hear what your experience was.

**Comment by montymxb** [comment]
Hey @ballcoach12 , we're actually looking into this internally right now within TypeFox. It's too early to share anything, but we're also interested in getting to a point where LLM support for DSLs (particularly Langium ones) can be defined in a standardized process. However, if you're interested (or others for that matter) you can feel free to get in touch with us to talk more about that 🙂.

**Reply by ballcoach12** [reply]
Hi @montymxb, I'd be very interested in talking with you guys further about this. We would like to see if we can help out in some way and maybe apply the solution to our project.

**Reply by spoenemann** [reply, marked as answer]
Hi @ballcoach12, please use the contact address shown on https://github.com/TypeFox to get in touch. Even if we're in an early stage, it would be great to hear about your expectations for such an AI-based solution.

**Reply by ballcoach12** [reply]
@spoenemann will do.
